rest/data/host.go

```diff
@@ -241,10 +241,16 @@ func makeSpawnOptions(options *restmodel.HostRequestOptions, user *user.DBUser)
 	}
 
 	if options.NoExpiration {
-		if options.SleepScheduleOptions.IsZero() {
-			options.SleepScheduleOptions = host.GetDefaultSleepSchedule(user.Settings.Timezone)
+		var defaultTimeZone string
+		if options.TimeZone != "" {
```

Comment:
Following the pattern below for the schedule, could we check if the options.TimeZone is empty and just update it and use options.TimeZone
if options.TimeZone == "" {
    options.TimeZone = user.Settings.Timezone
}
.... options.TimeZone
Or would it be incorrect to modify the options value? If that's the case, out of curisoity, what makes that not the case for the Schedule below?

Reply:
The time zone defaulting is that the preference order is:

* If the user explicitly chooses a time zone when first setting or modifying their sleep schedule, use that (i.e. `options.TimeZone`)
* If the host already exists and the user didn't explicitly choose a time zone, try to keep the host's existing time zone if it has one.
* If it's specified, use user's Spruce preferences time zone.
* If none of the above managed to set a time zone, use Spruce's default time zone (Eastern Time).

It's a little awkward here because of `host.GetDefaultSleepSchedule`, which also needs a time zone - the default sleep schedule is 8am-8pm Mon-Fri, but doesn't specify which time zone it's in. `GetDefaultSleepSchedule` needs to know what is the time zone to make a valid schedule and is a standalone function, so it can't reference the options' existing timezone field to see if it's already set or not.

I changed it so that instead of a `GetDefaultSleepSchedule`, we have a `SetDefaultSleepSchedule` method. It's kind of an unusual pattern since the more common pattern is to make a constructor for initialization, but makes the defaulting logic more streamlined in case of partially-set options.